modules/javafx.controls/src/test/java/test/javafx/scene/control/TreeTableCellTest.java

```diff
@@ -708,6 +722,145 @@ public void testCellInUneditableColumnIsNotEditable() {
+
+        cell.updateTreeTableView(tree);
+        cell.updateTreeTableColumn(editingColumn);
+    }
+
+    /**
+     * Sanity test: passes before fix of JDK-8187229, must pass after.
+     */
+    @Test
+    public void testEditCancelEventAfterCancelOnCell() {
+        setupForEditing();
+        int editingIndex = 1;
+        cell.updateIndex(editingIndex);
+        tree.edit(editingIndex, editingColumn);
+        TreeTablePosition<?,?> editingPosition = tree.getEditingCell();
+        List<CellEditEvent<?, ?>> events = new ArrayList<>();
+        editingColumn.setOnEditCancel(events::add);
+        cell.cancelEdit();
+        assertEquals("column must have received editCancel", 1, events.size());
+        assertEquals("editing location of cancel event", editingPosition, events.get(0).getTreeTablePosition());
+    }
+
+    @Test
+    public void testEditCancelEventAfterCancelOnTreeTable() {
+        setupForEditing();
+        int editingIndex = 1;
+        cell.updateIndex(editingIndex);
+        tree.edit(editingIndex, editingColumn);
+        TreeTablePosition<?, ?> editingPosition = tree.getEditingCell();
+        List<CellEditEvent<?, ?>> events = new ArrayList<>();
+        editingColumn.setOnEditCancel(events::add);
+        tree.edit(-1, null);
+        assertEquals("column must have received editCancel", 1, events.size());
+        assertEquals("editing location of cancel event", editingPosition, events.get(0).getTreeTablePosition());
+    }
+
+    /**
+     * Sanity test: passes before fix of JDK-8187229, must pass after.
+     */
+    @Test
+    public void testEditCancelEventAfterCellReuse() {
+        setupForEditing();
+        int editingIndex = 1;
+        cell.updateIndex(editingIndex);
+        tree.edit(editingIndex, editingColumn);
+        TreeTablePosition<?, ?> editingPosition = tree.getEditingCell();
+        List<CellEditEvent<?, ?>> events = new ArrayList<>();
+        editingColumn.setOnEditCancel(events::add);
+        cell.updateIndex(0);
+        assertEquals("column must have received editCancel", 1, events.size());
+        assertEquals("editing location of cancel event", editingPosition, events.get(0).getTreeTablePosition());
+    }
+
+    @Test
+    public void testEditCancelEventAfterCollapse() {
+        setupForEditing();
+        stageLoader = new StageLoader(tree);
+        int editingIndex = 1;
+        tree.edit(editingIndex, editingColumn);
+        TreeTablePosition<?, ?> editingPosition = tree.getEditingCell();
+        List<CellEditEvent<?, ?>> events = new ArrayList<>();
+        editingColumn.setOnEditCancel(events::add);
+        root.setExpanded(false);
+        Toolkit.getToolkit().firePulse();
+        assertEquals("column must have received editCancel", 1, events.size());
+        assertEquals("editing location of cancel event", editingPosition, events.get(0).getTreeTablePosition());
+    }
+
+    @Test
+    public void testEditCancelEventAfterModifyItems() {
```

Comment:
Does it make sense to have the same javadoc here as the same test in TableCellTest has?